share/dkg/pedersen/dkg.go

```diff
@@ -164,7 +177,14 @@ func NewDistKeyHandler(c *Config) (*DistKeyGenerator, error) {
 		canIssue = true
 	} else if !isResharing && newPresent {
 		// fresh DKG case
-		secretCoeff := c.Suite.Scalar().Pick(c.Suite.RandomStream())
+		randomStream := random.New()
+		// if the user provided a reader, use it alone or combined with crypto/rand
+		if c.Reader != nil && !c.UserReaderOnly {
```

Comment:
I'm not sure to get why you use the `UserReaderOnly` combined with the actual reader so that it doesn't mix with _crypto/random_. Wouldn't it be better to use _crypto/random_ by default when the reader is nil and use *only* the reader when it is defined ? Is there a security concern to mix it with _crypto/random_ ?

Reply:
We are not sure that the entropy given by the user is "good enough". So for security, the reader is mixed with crypto/rand by default to avoid any issue coming from bad randomness.

Reply:
As far as you provide your own source of entropy, it's your job to make sure it's a good one IMHO.

Reply:
I agree with you, but we cannot test that the randomness is good, so what we do here if that we make sure to always obtain good randomness no matter what the user supplies

Reply:
@Gilthoniel  The goal is to be able to always use a safe source of entropy when doing the DKG. It could have been done in drand, but that would have required (1) a method to mix a reader with crypto/rand - and there's a huge design space to explore so not necessarily trivial and (2) an additional action from the user of DKG before using it. 
I proposed that solution to @PizzaWhisperer because it doesn't involve any additional API to be safe (simply omitting the fields in the config will give you  a safe default) while allowing reproducible tests and external entropy sources.
At this point, we're on the realm of opinions in my opinion, but I'd like to think that this way of doing has more advantages than consequences. Happy to be proven wrong though !